tests/e2e/standalone/standalone_test.go

```diff
@@ -430,6 +430,33 @@ func testList(t *testing.T) {
 		require.NoError(t, err, "dapr stop failed")
 		assert.Contains(t, output, "app stopped successfully: dapr_e2e_list")
 	}, "run", "--app-id", "dapr_e2e_list", "-H", "3555", "-G", "4555", "--", "bash", "-c", "sleep 10 ; exit 0")
+
+	t.Run("daprd instance in list", func(t *testing.T) {
+		homeDir, err := os.UserHomeDir()
+		require.NoError(t, err)
+
+		path := filepath.Join(homeDir, ".dapr")
+		binPath := filepath.Join(path, "bin")
+		daprdPath := filepath.Join(binPath, "daprd")
+
+		if runtime.GOOS == "windows" {
+			daprdPath += ".exe"
+		}
+
+		cmd := exec.Command(daprdPath, "--app-id", "dapr_e2e_list", "--dapr-http-port", "3555", "--dapr-grpc-port", "4555", "--app-port", "0")
+		cmd.Start()
+
+		output, err := spawn.Command(getDaprPath(), "list")
+		t.Log(output)
+		require.NoError(t, err, "dapr list failed with daprd instance")
+		listOutputCheck(t, output)
+
+		output, err = spawn.Command(getDaprPath(), "stop", "--app-id", "dapr_e2e_list")
```

Comment:
So the stop command today actually talks to the CLI process that called `dapr run` . Since you are calling daprd directly, the stop command isn't actually shutting down your daprd instance. I believe why the tests are failing. There is a way to shutdown daprd cleanly and directly. Send an HTTP POST to http://localhost:3555/v1.0/shutdown to gracefully exit daprd. Alternatively, you can cmd.Kill()

Reply:
Doesn't the stop command code call the `dapr list` function (i.e. `List()`) in order to find the running `dapr` process and then kills the process with a matching `app id`? So, if the `List()` logic changes to include independent `daprd` processes, the stop command should also work on those processes, right? I had tested that locally before committing the tests and `dapr stop` also shuts down independent `daprd` processes. I'll add the direct shutdown to make sure that it works, though

Reply:
Specifically `dapr stop` is meant to be used in conjunction with `dapr run` for two reasons:
1. The dapr CLI process is the only one that knows the pid of the app that was spawned in addition to the daprd process. If we just kill daprd, there's the potential to orphan the app you were running with `dapr run`
2. On windows, there is no way to kill a proc that you don't own gracefully. So we signal the cli, which kills the processes it spawned.

We can change this behavior, but we have to make sure those cases are covered. And that it works on all platforms.

Reply:
@sk593 Have the scenarios mentioned above been tested in Windows?

Reply:
Yes, the independent daprd instances are picked up in Windows